Lib/imghdr.py

```diff
@@ -8,10 +8,13 @@
 # Recognize image headers #
 #-------------------------#
 
-def what(file, h=None):
+def what(file='', h=None):
```

Comment:
To catch 'what()' properly, the default should be a sentinel that people do not pass.  Before this add

```suggestion
_none = object()

def what(file=_none, h=None):
```